velox/functions/prestosql/tests/ProbabilityTest.cpp

```diff
@@ -155,5 +155,48 @@ TEST_F(ProbabilityTest, binomialCDF) {
       binomialCDF(-2, 0.5, -1), "numberOfTrials must be greater than 0");
 }
 
+TEST_F(ProbabilityTest, invBetaCDF) {
+  const auto invBetaCDF = [&](std::optional<double> a,
+                              std::optional<double> b,
+                              std::optional<double> p) {
+    return evaluateOnce<double>("inverse_beta_cdf(c0, c1, c2)", a, b, p);
+  };
+
+  EXPECT_EQ(0.0, invBetaCDF(3, 3.6, 0.0));
+  EXPECT_EQ(1.0, invBetaCDF(3, 3.6, 1.0));
+  EXPECT_EQ(0.34696754854406159, invBetaCDF(3, 3.6, 0.3));
+  EXPECT_EQ(0.76002724631002683, invBetaCDF(3, 3.6, 0.95));
+
+  EXPECT_EQ(std::nullopt, invBetaCDF(std::nullopt, 3.6, 0.95));
+  EXPECT_EQ(std::nullopt, invBetaCDF(3.6, std::nullopt, 0.95));
+  EXPECT_EQ(std::nullopt, invBetaCDF(3.6, 3.6, std::nullopt));
+
+  // Boost libraries currently throw an assert. Created the expected values via
+  // Matlab. Presto currently throws an exception from Apache Math for these
+  // values
+  // EXPECT_EQ(0.5, invBetaCDF(kDoubleMax, kDoubleMax, 0.3));
```

Comment:
This is a good catch. For these values I feel that we can write test calls that validate the assertion thrown from Boost with VELOX_ASSERT_THROW. Also just state that Presto throws an exception from Apache Math for the same values. 

We could state the Matlab behavior also. The Presto/Velox groups don't look at Matlab in general. But its interesting that it returns a successful value.

Reply:
Unfortunately `VELOX_ASSERT_THROW` does not work at catching the assert. The test case just hangs here and has to be killed.
```
[ RUN      ] ProbabilityTest.invBetaCDF
Assertion failed: (*p_derivative >= 0), function ibeta_fraction2, file beta.hpp, line 726.
```
When I added `VELOX_ASSERT_THROW(invBetaCDF(kDoubleMax, kDoubleMax, 0.3), "(*p_derivative >= 0)");`

This is I think due to it being an Assert rather than an exception as VELOX_ASSERT_THROW seems to catch exceptions and I am unaware of a way of catching an assert.

```
#define VELOX_ASSERT_THROW(expression, errorMessage)                 \
  try {                                                              \
    (expression);                                                    \
    FAIL() << "Expected an exception";                               \
  } catch (const VeloxException& e) {                                \
    ASSERT_TRUE(e.message().find(errorMessage) != std::string::npos) \
        << "Expected error message to contain '" << errorMessage     \
        << "', but received '" << e.message() << "'.";               \
  }
```

Assert is from 
`BOOST_ASSERT(*p_derivative >= 0);` in boost's beta.hpp

Reply:
Yes, this is the conventional assert. There wouldn't be a way of catching it. But its strange that the test hangs. I would've expected it to crash.